src/modules/sync/lib/ipc-handlers.ts

```diff
@@ -158,16 +166,94 @@ export async function pushArchive(
 		throw new Error( 'No token found' );
 	}
 
+	let isUploadingPaused = false;
+	const file = fs.createReadStream( archivePath );
+	const fileSize = fs.statSync( archivePath ).size;
+	const filename = path.basename( archivePath );
+
+	const cleanup = () => {
+		file.destroy();
+		file.close();
+	};
+
+	const attachmentId = await new Promise< string >( ( resolve, reject ) => {
+		const upload = new Upload( file, {
+			endpoint: `${ STUDIO_FILE_UPLOADS_ENDPOINT_BASE }/${ remoteSiteId }`,
+			chunkSize: STUDIO_FILE_UPLOADS_CHUNK_SIZE,
+			retryDelays: STUDIO_FILE_UPLOADS_RETRY_DELAYS,
+			overridePatchMethod: true,
+			removeFingerprintOnSuccess: true,
+			storeFingerprintForResuming: true,
+			headers: {
+				Authorization: `Bearer ${ token.accessToken }`,
+			},
+			metadata: {
+				filename,
+				filetype: 'application/gzip',
+			},
+			uploadSize: fileSize,
+			onBeforeRequest: ( req: HttpRequest ) => {
```

Comment:
```suggestion
			onBeforeRequest: ( req ) => {
```
`tus-js-client` already provides the type.

Reply:
Updated in [c1efbe1](https://github.com/Automattic/studio/pull/2212/commits/c1efbe13c0b111cb0f49e74ed76bc9f3eaffccf2).